airflow/contrib/operators/kubernetes_operator.py

```diff
@@ -295,6 +295,19 @@ def poll_job_completion(self, job_name, dependent_containers={'cloudsql-proxy'})
                         if live_cnt > 0:
                             has_live = True
                             break
+                    elif 'Succeeded' == pod['status']['phase']:
+                        # For us to end up in this block, the job has to be Running and the pod has to be Succeeded.
+                        # This happens when (on a previous attempt) we successfully finished execution, killed dependent
+                        # containers, and failed to delete the job.
+                        # In this scenario, we want to immediately stop polling, and retry job deletion.
+                        has_live_existed = True
+                        has_live = False
+                    else:
+                        raise Exception(
```

Comment:
There are 5 states we should be handling ([docs](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.16/#podstatus-v1-core), [more docs](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase)):
* **Pending:** The pod has been accepted by the Kubernetes system, but one or more of the container images has not been created. This includes time before being scheduled as well as time spent downloading images over the network, which could take a while. 
* **Running:** The pod has been bound to a node, and all of the containers have been created. At least one container is still running, or is in the process of starting or restarting.
* **Succeeded:** All containers in the pod have terminated in success, and will not be restarted. 
* **Failed:** All containers in the pod have terminated, and at least one container has terminated in failure. The container either exited with non-zero status or was terminated by the system.
* **Unknown:** For some reason the state of the pod could not be obtained, typically due to an error in communicating with the host of the pod.

We should explicitly handle failed rather than fall through to the `else` condition.